### PR TITLE
Add service principals by app id microsoft.graph rbac

### PR DESCRIPTION
**Pull request by ankushbindlish2**
### Latest improvements:
<i>MSFT employees can try out our new experience at <b>[OpenAPI Hub](https://aka.ms/openapiportal) </b> - one location for using our validation tools and finding your workflow. 
</i><br>
### Contribution checklist:
- [x] I have reviewed the [documentation](https://github.com/Azure/azure-rest-api-specs#basics) for the workflow.
- [x] [Validation tools](https://github.com/Azure/azure-rest-api-specs/blob/master/documentation/swagger-checklist.md#validation-tools-for-swagger-checklist) were run on swagger spec(s) and have all been fixed in this PR.
- [x] The [OpenAPI Hub](https://aka.ms/openapiportal) was used for checking validation status and next steps.
### ARM API Review Checklist
- [x] Service team MUST add the "WaitForARMFeedback" label if the management plane API changes fall into one of the below categories. 
- adding/removing APIs.
- adding/removing properties.
- adding/removing API-version. 
- adding a new service in Azure.

Failure to comply may result in delays for manifest application. Note this does not apply to data plane APIs.
- [ ] If you are blocked on ARM review and want to get the PR merged urgently, please get the ARM oncall for reviews (RP Manifest Approvers team under Azure Resource Manager service) from IcM and reach out to them. 
Please follow the link to find more details on [API review process](https://armwiki.azurewebsites.net/rp_onboarding/ResourceProviderOnboardingAPIRevieworkflow.html).

Previous PR: https://github.com/Azure/azure-rest-api-specs/pull/5387
Why we are doing this ? 
This service already exposed this method and it should be safe to add into swagger.
We are trying to get Application Object for a guest user with default privileges.

Example : GET /72f988bf-86f1-41af-91ab-2d7cd011db47/servicePrincipalsByAppId/9469b9f5-6722-4481-a2b2-14ed560b706f/objectId?api-version=1.6 HTTP/1.1
{"odata.metadata":"https://graph.windows.net/72f988bf-86f1-41af-91ab-2d7cd011db47/$metadata#Edm.String","value":"3eb9d9f3-371c-4d24-a60a-d7681c48c071"}

https://docs.microsoft.com/en-us/previous-versions/azure/ad/graph/api/functions-and-actions#serviceprincipalsbyappid-get-service-principal-object-id-by-application-id--

### CODE REVIEW

**Comment by adxsdknet** [comment]
# Automation for azure-sdk-for-net 
A PR has been created for you: 
https://github.com/azure/azure-sdk-for-net/pull/5495 
.NET SDK Commits: 
https://github.com/adxsdknet/azure-sdk-for-net/pull/5495/commits/56d5f6ddb4ecc5f91bebca0d8263cfb886830614

File: specification/graphrbac/data-plane/Microsoft.GraphRbac/stable/1.6/graphrbac.json [hunk 1]
```diff
@@ -3191,6 +3225,20 @@
       },
       "description": "Application list operation result."
     },
+    "ServicePrincipalObjectResult": {
+      "type": "object",
+      "properties": {
+        "value": {
+          "type": "string",
+          "description": "Service Principal Application Object Id"
+        },
```
Comment (darshanhs90) [inline specification/graphrbac/data-plane/Microsoft.GraphRbac/stable/1.6/graphrbac.json]: isnt value an array of strings?
{
  "odata.metadata": "https://graph.windows.net/myorganization/$metadata#Edm.String",
  "value": [
    "00b4e797-7017-4720-b187-b01981c820d6"
  ]
}

**Comment by jhendrixMSFT** [comment]
@AutorestCI regenerate azure-sdk-for-go

**Review by grlin — APPROVED** [review]

**Comment by sergey-shandar** [comment]
@darshanhs90 are you ok with the changes?

File: specification/graphrbac/data-plane/Microsoft.GraphRbac/stable/1.6/graphrbac.json [hunk 2]
```diff
@@ -1284,6 +1284,40 @@
         "x-ms-odata": "#/definitions/ServicePrincipal"
       }
     },
+    "/{tenantID}/servicePrincipalsByAppId/{applicationID}/objectId": {
```
Comment (lmazuel) [inline specification/graphrbac/data-plane/Microsoft.GraphRbac/stable/1.6/graphrbac.json]: Not a comment, just ref for me/
https://docs.microsoft.com/en-us/previous-versions/azure/ad/graph/api/functions-and-actions#servicePrincipalsByAppId

File: specification/graphrbac/data-plane/Microsoft.GraphRbac/stable/1.6/graphrbac.json [hunk 3]
```diff
@@ -3191,6 +3225,24 @@
...
       "description": "Application list operation result."
     },
+    "ServicePrincipalObjectResult": {
+      "type": "object",
+      "properties": {
+        "value": {
+          "type": "array",
+          "items": {
+            "type": "string",
+            "description": "Service Principal Application Object Id"
```
Comment (lmazuel) [inline specification/graphrbac/data-plane/Microsoft.GraphRbac/stable/1.6/graphrbac.json]: Let's use the description from the doc:
`The Object ID of the service principal with the specified application ID.`

**Review by darshanhs90 — APPROVED** [review]

File: specification/graphrbac/data-plane/Microsoft.GraphRbac/stable/1.6/graphrbac.json [hunk 4]
```diff
@@ -1284,6 +1284,40 @@
...
+        "description": "Gets an object id for a given application id from the current tenant.",
+        "parameters": [
+          {
+            "$ref": "#/parameters/ApiVersionParameter"
+          },
+          {
+            "$ref": "#/parameters/tenantIDInPath"
+          },
+          {
+            "$ref": "#/parameters/applicationIdInPath"
```
Comment (lmazuel) [inline specification/graphrbac/data-plane/Microsoft.GraphRbac/stable/1.6/graphrbac.json]: For consistency with the rest of the file, please just inline this:
```json
          {
            "name": "applicationObjectId",
            "in": "path",
            "required": true,
            "type": "string",
            "description": "Application object ID."
          },
```

Comment (lmazuel) [inline specification/graphrbac/data-plane/Microsoft.GraphRbac/stable/1.6/graphrbac.json]: Please inline it (like the rest of the file) and call it simply `applicationId` (no InPath)

File: specification/graphrbac/data-plane/Microsoft.GraphRbac/stable/1.6/graphrbac.json [hunk 5]
```diff
@@ -1284,6 +1284,40 @@
         "x-ms-odata": "#/definitions/ServicePrincipal"
       }
     },
+    "/{tenantID}/servicePrincipalsByAppId/{applicationID}/objectId": {
+      "get": {
+        "tags": [
+          "ServicePrincipalsByAppId"
+        ],
+        "operationId": "ServicePrincipalsByAppId_Get",
```
Comment (lmazuel) [inline specification/graphrbac/data-plane/Microsoft.GraphRbac/stable/1.6/graphrbac.json]: I would suggest:
`Applications_getServicePrincipalsIdByAppId`

File: specification/graphrbac/data-plane/Microsoft.GraphRbac/stable/1.6/graphrbac.json (hunk 3 above)
Comment (yugangw-msft) [reply specification/graphrbac/data-plane/Microsoft.GraphRbac/stable/1.6/graphrbac.json]: Why it is an array? I thought each app should only have a single service principal in the same tenant. Also, the description is `Gets an object id for a given application id from the current tenant.`, which to me means a singular value

Comment (ankushbindlish2) [reply specification/graphrbac/data-plane/Microsoft.GraphRbac/stable/1.6/graphrbac.json]: Service json format is array. Please observe example above.

This was incorporated as a comment resolution. See above @hari comment

File: specification/graphrbac/data-plane/Microsoft.GraphRbac/stable/1.6/graphrbac.json (hunk 5 above)
Comment (ankushbindlish2) [reply specification/graphrbac/data-plane/Microsoft.GraphRbac/stable/1.6/graphrbac.json]: Will incorporate

File: specification/graphrbac/data-plane/Microsoft.GraphRbac/stable/1.6/graphrbac.json (hunk 4 above)
Comment (ankushbindlish2) [reply specification/graphrbac/data-plane/Microsoft.GraphRbac/stable/1.6/graphrbac.json]: Following tenantId naming convention. Won't fix for now.

**Comment by sergey-shandar** [comment]
@lmazuel Are you ok with the changes?

File: specification/graphrbac/data-plane/Microsoft.GraphRbac/stable/1.6/graphrbac.json (hunk 4 above)
Comment (lmazuel) [reply specification/graphrbac/data-plane/Microsoft.GraphRbac/stable/1.6/graphrbac.json]: I insist, please inline this here and not ref an external definition to follow the consistency in this file. Thank!

**Review by lmazuel — APPROVED** [review]
I actually tested the Python code from this, and it works

**Pull request merged**